server.py

```diff
@@ -98,21 +154,20 @@ def main():
                 if len(msg_chunk) < 1024:
                     try:
                         resp_uri = parse_request(msg)
+                        response = response_ok(resp_uri)
```

Comment:
By modifying the `response_ok` function to wrap itself around a call to `resolve_uri` you make it so that it is not possible to test the one without engaging the other.  This violates the principle that each function should perform one and only one task.

The `response_ok` function should be responsible for building and returning a `200 OK` HTTP response only.  The `resolve_uri` method should be responsible for getting the resource identified by a URI.  If you must chain the calls, do so like this instead:

```python
response = response_ok(resolve_uri(resp_uri))
```

See how that allows the two functions to remain separate from each-other, testable in isolation?